MoreLinq/EndsWith.cs

```diff
@@ -71,8 +71,17 @@ public static bool EndsWith<T>(this IEnumerable<T> first, IEnumerable<T> second,
 
             comparer = comparer ?? EqualityComparer<T>.Default;
 
+            var nullableSecondCount = second.TryGetCollectionCount();
+
+            if (nullableSecondCount is int secondCount &&
```

Comment:
This starting condition here is the same as the one on the `return` statement below so I don't think it has to be tested twice. What about collapsing it so?

```c#
            List<T> secondList;
            return second.TryGetCollectionCount() is int secondCount
                 ? first.TryGetCollectionCount() is int firstCount && secondCount > firstCount
                   ? false
                   : Impl(second, secondCount)
                 : Impl(secondList = second.ToList(), secondList.Count);
```

The benefit is that `nullableSecondCount` and `collectionCount` (which is the same as `secondCount`) are not needed.